user/pages/05.blog/2018-01-09-docker-improvements/item.md

```diff
@@ -0,0 +1,171 @@
+
+```
+volumes:
+  - ./etc-overlay:/opt/opennms-etc-overlay
+```
+
+ATTENTION: You have to be aware, about configuration files changed in the Web UI and which are still persisted on the files system.
+           This files need to be maintained, some examples are users.xml, groups.xml, snmp-config.xml, surveillance-views.xml, etc..
+           Ideally this configuration files should be in the database but there is currently not really another useful workaround for this.
+           You have to make sure your `${OPENNMS_HOME}/etc` directory gets persisted in a local volume.
+           The Karaf cache directory is by default not persisted and gets lost on restart.
+           If you work with the Topology UI ensure the `${OPENNMS_HOME}/data` gets persisted in a local volume as well, you have to deal with this during updates!
+
+## Image Tags
+
+We have cleaned up the image tags to provide the following use cases:
+
+* specific stable release version, e.g. `21.0.2`: Use this tag if you have a specific version dependencies in your service stack and you want full control for upgrades
+* floating stable release tag `latest`: Use this tag when you want to run always the latest stable release, update the container image to latest stable release with `docker-compose pull && docker-compose stop opennms && docker-compose up -d`
+* floating daily snapshot tag `bleeding`: This is an image built on a daily base from the latest `develop` branch and is the release candidate for the next major release of horizon
+* feature tag `drift`: This is a special tag for daily snapshot built if you want to test the features which are currently developed in [Project Drift](https://wiki.opennms.org/wiki/DevProjects/Drift). This tag will be removed when this feature is merged in a released version of Horizon.
+
+## Allow Custom Health checks
+
+In previous container image versions the Health check where hard coded.
+To give people more freedom to implement container health checks, it is moved out of the container image into the `docker-compose.yml` file where the service stack is described.
+In case you want to run the Java web application and the core services in separate containers, you can now define different health checks for each service.
+
+## Service Dependencies
+
+The old OpenNMS Horizon image had a sleep waiting for the PostgreSQL database.
+With Docker Compose version 2.1 and later it is possible to wait for a service condition.
+It makes the wait obsolete and OpenNMS Horizon is started when the PostgreSQL database is in state "healthy".
+
+## Java Process and PID 1
+
+Processes with ID 1 have special functionalities related to signal handling and cleaning up zombie processes.
+Using a current the example `docker-compose.yml` uses now the built-in `init: true` option, which runs an tiny init inside the container and prevents the Java process to get PID 1.
```

Comment:
For example, `docker-compose.yml` now uses the built-in `init: true` option, which runs a tiny init inside the container and prevents the Java process from getting PID 1.